### PR TITLE
ci: shift automatic checks to ease debugging

### PR DESCRIPTION
**Pull request by sadym-chromium**
1. Move `unused variable` check from ts build to eslint. For debugging purposes, often commenting out pieces of code is required. Sometimes it leads to build failures, like `error    'foo' is defined but never used`.
2. Move pre-commit checks from `pre-commit` to `pre-push` stage. This is required to be able to fix a progress without polishing out the whole project before each commit + to speed up the commits.

### CODE REVIEW

**Review by OrKoN — COMMENTED** [review]
This does not look like a feature. I suggest `test:` or `chore:`

File: README.md [hunk 1]
```diff
@@ -153,6 +153,10 @@ We use [cddlconv](https://github.com/google/cddlconv) to generate our WebDriverB
 
 Refer to the documentation at [.pre-commit-config.yaml](.pre-commit-config.yaml).
 
+```sh
+pre-commit install pre-push
```
Comment (thiagowfx) [inline README.md:157]: don't we need `--hook-types` here?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@
 #
 # Apply all pre-commit hooks to all files:
 #   $ pre-commit run --all-files
-default_stages: [pre-commit]
+default_stages: [pre-push]
```
Comment (thiagowfx) [inline .pre-commit-config.yaml:23]: Does this work as expected? I understand the intent, but please test it manually, as pre-commit configs often have unintuitive behavior.

**Comment by thiagowfx** [comment]
`ci:` is better IMHO

File: README.md (hunk 1 above)
Comment (sadym-chromium) [reply README.md]: done. Thanks!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (sadym-chromium) [reply .pre-commit-config.yaml:23]: The check didn't allow me pushing [commit with an unused function](https://github.com/GoogleChromeLabs/chromium-bidi/pull/1421/commits/eeb80f58c0555b1aaff0b39343ac41c190b578c6). Eslint failed:
```
...
|eslint...................................................................Failed
- hook id: eslint
- exit code: 1

.../chromium-bidi/src/bidiServer/index.ts
  23:10  error    'foo' is defined but never used  @typescript-eslint/no-unused-vars
  23:16  warning  Unexpected empty function 'foo'  @typescript-eslint/no-empty-function

✖ 3 problems (1 error, 2 warnings)
```

Comment (thiagowfx) [reply .pre-commit-config.yaml:23]: LGTM then

**Review by thiagowfx — APPROVED** [review]

**Pull request merged**